src/index/index_factory.cc

```diff
@@ -93,12 +102,31 @@ IndexFactory::MapInstance() {
     static FuncMap func_map;
     return func_map;
 }
+
+IndexFactory::FeatureMap&
+IndexFactory::FeatureMapInstance() {
+    static FeatureMap featureMap;
+    return featureMap;
+}
+
 IndexFactory::GlobalIndexTable&
 IndexFactory::StaticIndexTableInstance() {
     static GlobalIndexTable static_index_table;
     return static_index_table;
 }
 
+bool
+IndexFactory::FeatureCheck(const std::string& name, uint64_t feature) const {
```

Comment:
Is this for external usage? If so, why why we make it in factory?

Reply:
This consideration is mainly to pass the index-related features to the Milvus side when registering the index. This way, Milvus can use the feature associated with the indexType to make branching decisions, rather than handling things directly based on indexType. The reason for putting it in the factory is to bind the feature bitset with indexType, simplifying the registration process. Do you have any suggestions here?

Reply:
Should we bind it with version system? Since the feature supports vary from version to version